spec/models/user.spec.js

```diff
@@ -1,58 +1,165 @@
 const mongoose = require("mongoose");
 const bcrypt = require('bcrypt');
-
 require("../mongodb_helper");
 const User = require("../../models/user");
 
+const createAndValidateUser = async(obj) => {
+  const user = new User(obj);
+  await user.validate();
+  return user;
+};
+
 describe("User model", () => {
   beforeEach((done) => {
     mongoose.connection.collections.users.drop(() => {
       done();
     });
   });
 
-  it("can list several users", async (done) => {
-    const user = new User({
-      username: "peter",
-      email: "someone@example.com",
-      password: "password",
-    });
 
-    const user2 = new User({
-      username: "bob",
-      email: "someone2@example.com",
-      password: "pass2word",
+  it("requires email to be unique", async () => {
+    const existingUser1 = new User({
+      email: "existing@example.com",
+      password: "password123",
+      username: "existinguser1",
     });
+  
+    // Create another user with the same email
+    const existingUser2 = new User({
+      email: "existing@example.com",
+      password: "password456",
+      username: "existinguser2",
+    });
+  
+  
+    await existingUser1.save();
+ 
+    try {
+      await existingUser2.save();
+    } catch (error) {
+      expect(error.errors["email"].message).toBe("Email address is already in use");
```

Comment:
not sure any of these catch tests are being wrong

Reply:
I've pushed again if that's showing ? Still have 2 failing tests but wonder if someone else wouldnt mind taking a look as my head hurts.